### PR TITLE
Product details

### PR DESCRIPTION
**Pull request by sr-meh**
# Description 

## Issue [38](https://github.com/BrightCoders-Institute/proyecto-final-jdg/issues/38)

- Modify the way the products are displayed on the main page. Now the products stored in the database are displayed correctly.

- Add an 'Add to cart' button on the cards on the main page.

- Add styles to the product detail page.

- Adds a different view for admins and another view for customers on the product details page.

- Adds validation to allow both users with an active session and users without a session to view the product details.

### Customer view

image
image

### Admin view 
image

### CODE REVIEW

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -89,5 +89,18 @@ body {
...
+
+.product img {
+  height: 400px;
+  max-width: 100%;
+}
+
+.product h2 {
+  margin-top: 10px;
+  font-size: 1.2em;
+}
```
Comment (JuanequeX) [inline app/assets/stylesheets/application.scss:106]:
```suggestion
.product {
  border: 1px solid #ddd;
  padding: 15px;
  text-align: center;

  img {
    height: 400px;
    max-width: 100%;
  }

  h2 {
    margin-top: 10px;
    font-size: 1.2em;
  }
}

```

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -2,17 +2,19 @@ class ApplicationController < ActionController::Base
...
   def customer?
-    unless current_user.usertype == 'customer'
-      redirect_to products_path
+    if current_user && current_user.usertype != 'customer'
+      puts "Redirecting to products_path"
+      redirect_to products_path unless action_name == 'show'
     end
   end
 
+
```
Comment (JuanequeX) [inline app/controllers/application_controller.rb:17]: remove this line

File: app/controllers/products_controller.rb [hunk 3]
```diff
@@ -84,7 +85,7 @@ def destroy
     end
   end
 
-  
+
```
Comment (JuanequeX) [inline app/controllers/products_controller.rb:89]: remove these extra lines

File: app/views/products/show.html.erb [hunk 4]
```diff
@@ -1,81 +1,123 @@
...
+            </tr>
+          </thead>
+          <tbody>
+            <tr>
+              <td class="align-middle">Image</td>
+              <td>
+                <% if @product.image.attached? %>
+                  <%= image_tag rails_blob_path(@product.image, disposition: "attachment", only_path: true), class: "img-thumbnail img-fluid", width: "100", height: "100" %>
+                <% else %>
+                  No Image
```
Comment (JuanequeX) [inline app/views/products/show.html.erb:21]: Enhancing the user experience involves adding a default system image rather than using plain text.

**Review by JuanequeX — CHANGES_REQUESTED** [review]

**Review by PudinDeveloper — APPROVED** [review]

**Review by JuanequeX — APPROVED** [review]

**Review by juperameza — APPROVED** [review]
Lgtm

**Pull request merged**